service/collector/app_operator.go

```diff
@@ -130,42 +146,79 @@ func (a *AppOperator) collectAppVersions(ctx context.Context) (map[string]bool,
 
 	for _, app := range l.Items {
 		version := app.Labels[label.AppOperatorVersion]
-		appVersions[version] = true
+		appNamespaces, ok := appVersions[version]
+		if !ok {
+			namespace := app.Namespace
+
+			v, err := semver.NewVersion(version)
+			if err != nil {
+				a.logger.Errorf(ctx, err, "failed to parse version %#q", version)
+				continue
+			}
+			if v.Major() < 4 {
```

Comment:
This is going to be `giantswarm`, even if we don't do this, right?

Reply:
This gets a list of the app versions and which namespaces we need to check.

So there is a matching app-operator for the namespace. If the app CR version is < 4.0.0 the app-operator is in the giantswarm namespace.

For app-operator >= 4.0.0 it will be the cluster namespace plus `0.0.0` which matches to app-operator-unique in thw giantswarm namespace.

So this part is needed but its a bit tricky. Let me know if it needs more comments.